charon-ml/src/Expressions.ml

```diff
@@ -1,27 +1,75 @@
+(** WARNING: this file is partially auto-generated. Do not edit `src/Expressions.ml`
+    by hand. Edit `templaces/Expressions.ml` instead, or improve the code
+    generation tool so avoid the need for hand-writing things.
+
+    `templaces/Expressions.ml` contains the manual definitions and some `(*
+    __REPLACEn__ *)` comments. These comments are replaced by auto-generated
+    definitions by running `make generate-ml` in the crate root. The
+    code-generation code is in `charon/src/bin/generate-ml`.
+ *)
 open Identifiers
+
 open Types
 open Values
 module VarId = IdGen ()
 module GlobalDeclId = Types.GlobalDeclId
 module FunDeclId = Types.FunDeclId
 
-type fun_decl_id = FunDeclId.id [@@deriving show, ord]
-
 (** We define this type to control the name of the visitor functions
     (see e.g., {!Charon.UllbcAst.iter_statement_base}).
   *)
 type var_id = VarId.id [@@deriving show, ord]
 
+(** An assumed function identifier, identifying a function coming from a
+ standard library. *)
 type assumed_fun_id =
-  | BoxNew
+  | BoxNew  (** `alloc::boxed::Box::new` *)
   | BoxFree
+      (** `alloc::alloc::box_free`
+ This is actually an unsafe function, but the rust compiler sometimes
```

Comment:
We should add indentation in multi-line comments

Reply:
The `--wrap-comments` ocamlformat option doesn't work on doc-comments and destroys the formatting of e.g. code blocks

Reply:
I can still guess the indent level, that'd be better than the current state.

Reply:
This worked great (https://github.com/AeneasVerif/charon/pull/307)